src/App.jsx

```diff
@@ -14,45 +17,95 @@ export const goodsFromServer = [
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'letter';
+const SORT_BY_LENGTH = 'length';
+
+const sortingBy = (goods, property) => {
+  const sortedGoods = [...goods];
+
+  switch (property) {
+    case SORT_ALPHABETICALLY:
+      sortedGoods.sort((item1, item2) => item1.localeCompare(item2));
+      break;
+    case SORT_BY_LENGTH:
+      sortedGoods.sort((item1, item2) => item1.length - item2.length);
+      break;
+    default:
+      return 0;
+  }
+
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [sortingProperties, setSortingProperty] = useState(null);
+  const [reverseSorting, setReverseSorting] = useState(false);
+  let goods = [...goodsFromServer];
+
+  if (sortingProperties) {
+    goods = sortingBy(goods, sortingProperties);
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (reverseSorting) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortingProperties !== SORT_ALPHABETICALLY,
+          })}
+          onClick={() => setSortingProperty(SORT_ALPHABETICALLY)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortingProperties !== SORT_BY_LENGTH,
+          })}
+          onClick={() => setSortingProperty(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': reverseSorting !== true,
+          })}
+          onClick={
+            reverseSorting
+              ? () => setReverseSorting(false)
+              : () => setReverseSorting(true)
+          }
+        >
+          Reverse
+        </button>
+
+        {(sortingProperties || reverseSorting) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortingProperty(null);
+              setReverseSorting(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
+      </div>
+
+      <ul>
+        {goods.map(good => (
+          <li data-cy="Good">{good}</li>
+        ))}
```

Comment:
don't forget about key prop

Reply:
@etojeDenys eslinter doesn't allow to use array index as key.
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/33396860/edabcaf3-d0a6-4c34-8a95-e5129549f195)